Sources/StreamChat/Query/Filter.swift

```diff
@@ -99,18 +99,27 @@ public struct Filter<Scope: FilterScope> {
     /// The "right-hand" side of the filter. Specifies the value the filter should match.
     public let value: FilterValue
 
+    /// The mapper that will transform the input value to a value that
+    /// can be compared with the DB value
+    typealias ValueMapper = (Any) -> FilterValue?
+    let valueMapper: ValueMapper?
+
+    /// The keypath of the DB object that will be compared with the input value during
+    /// a local filtering.
     let keyPathString: String?
 
     init(
         operator: String,
         key: String?,
         value: FilterValue,
+        valueMapper: ValueMapper?,
```

Comment:
The problem I see with this approach is that it might be error prone, we need to make sure we don't forget to provide a value mapper for all enums or for every FilterValue that requires it. IMO we should make sure all FilterValues are convirtible to String raw value

Reply:
We had a discussion about this. The problem with having a way to convert everything to `String` it may require us to expose functionality to customers that it's not needed. 

In this way we are very explicit about when we need to include the valueMapper. 

As this approach is meant for internal usage only and helps us quickly resolve the issue we have, I consider it safe enough. Once we start exposing more entities into auto filtering and especially when we need to expose the functionaliy publically we can reevaluate it

Reply:
Exposing a filterRawValue should not be a big deal, but fine, if the team is okay with this approach. I just think we could introduce similar issues in the future. But we can improve it later if we verify it is error prone 👍

Reply:
Either way, we should test all filters to make sure they are converted correctly. At least only the equal operator. For example, how is URL converted?

Reply:
Exposing a `filterRawValue` means that we are moving the responsibility for converting the value to the filter when the definition of it is happening on the FilterKey. For simplicity, we should try to keep the filter as clean and unaware of the key and it's operations as possible. In that way we can have every related to the key and how it's being interpreted in one place.